pandas/tests/indexing/test_chaining_and_caching.py

```diff
@@ -435,6 +435,16 @@ def test_detect_chained_assignment_warnings_filter_and_dupe_cols(self):
             )
             tm.assert_frame_equal(df, expected)
 
+    @pytest.mark.parametrize("rhs", [3, DataFrame({0: [1, 2, 3, 4]})])
+    def test_detect_chained_assignment_warning_stacklevel(self, rhs):
+        # GH#42570
+        df = DataFrame(np.arange(25).reshape(5, 5))
+        chained = df.loc[:3]
+        with option_context("chained_assignment", "warn"):
+            with tm.assert_produces_warning(com.SettingWithCopyWarning) as t:
```

Comment:
i think `check_stacklevel` is *supposed* to do this?

Reply:
Yeah thought so too, but test was passing with stacklevel 4 and 5